src/Microsoft.DotNet.Build.Tasks/Targets/packageresolve.targets

```diff
@@ -23,9 +22,10 @@
     <ResolveNugetPackages>false</ResolveNugetPackages>
   </PropertyGroup>
 
-  <Target Name="RestorePackages" 
+  <!-- Restoring packages during a background (designtime) build will cause VS 2015 (v14) to get into an endless loop of resolving references. -->
+  <Target Name="RestorePackages"
           BeforeTargets="ResolveNuGetPackages"
-          Condition="'$(RestorePackages)'=='true'">
+          Condition="'$(RestorePackages)'=='true' and !('$(VSDesignTimeBuild)'=='true' and '$(VisualStudioVersion)' >= '14.0')">
```

Comment:
I would prefer that we actually use those extra conditions to set RestorePackages property instead of adding it here. We already do that for the DesignTimeBuild property already.